src/QMCDrivers/VMC/VMCBatched.cpp

```diff
@@ -2,68 +2,61 @@
 // See LICENSE file in top directory for details.
 //
-// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+// Copyright (c) 2019 QMCPACK developers.
 //
-// File developed by: Bryan Clark, bclark@Princeton.edu, Princeton University
-//                    Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
-//                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
-//                    Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
-//                    Cynthia Gu, zg1@ornl.gov, Oak Ridge National Laboratory
-//                    Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
-//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
 //
-// File created by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+// File refactored from: VMC.cpp
 //////////////////////////////////////////////////////////////////////////////////////
 
-
 #include "QMCDrivers/VMC/VMCBatched.h"
 #include "QMCDrivers/VMC/VMCUpdatePbyP.h"
 #include "QMCDrivers/VMC/VMCUpdateAll.h"
-#include "OhmmsApp/RandomNumberControl.h"
-#include "Message/OpenMP.h"
-#include "Message/CommOperators.h"
-#include "Utilities/RunTimeManager.h"
-#include <qmc_common.h>
-//#define ENABLE_VMC_OMP_MASTER
-#include "ADIOS/ADIOS_profile.h"
-#if !defined(REMOVE_TRACEMANAGER)
-#include "Estimators/TraceManager.h"
-#else
-typedef int TraceManager;
-#endif
 
 namespace qmcplusplus
 {
-/// Constructor.
-VMCBatched::VMCBatched(MCPopulation& pop,
+/** Constructor maintains proper ownership of input parameters
+   */
+VMCBatched::VMCBatched(QMCDriverInput& qmcdriver_input,
+                       VMCDriverInput& input,
+                       MCPopulation& pop,
                        TrialWaveFunction& psi,
                        QMCHamiltonian& h,
                        WaveFunctionPool& ppool,
                        Communicate* comm)
-    : QMCDriverNew(pop, psi, h, ppool, comm), UseDrift("yes")
+    : QMCDriverNew(qmcdriver_input, pop, psi, h, ppool, comm), vmcdriver_input_(input)
 {
-  RootName = "vmc";
-  QMCType  = "VMCBatched";
-  qmc_driver_mode.set(QMC_UPDATE_MODE, 1);
-  qmc_driver_mode.set(QMC_WARMUP, 0);
-  m_param.add(UseDrift, "useDrift", "string");
-  m_param.add(UseDrift, "usedrift", "string");
-  m_param.add(UseDrift, "use_drift", "string");
+  // qmc_driver_mode.set(QMC_UPDATE_MODE, 1);
+  // qmc_driver_mode.set(QMC_WARMUP, 0);
+}
 
-  prevSteps               = nSteps;
-  prevStepsBetweenSamples = nStepsBetweenSamples;
+VMCBatched::IndexType VMCBatched::calc_default_local_walkers()
+{
+  IndexType rw = vmcdriver_input_.get_requested_walkers_per_rank();
+  if (rw < num_crowds_)
+    rw = num_crowds_;
+  walkers_per_crowd_      = (rw % num_crowds_) ? rw / num_crowds_ + 1 : rw / num_crowds_;
```

Comment:
walkers_per_crowd should be a vector of a size equal to the number of crowds.
What you defined here seems to be the maximum number of walkers per crowd.

Reply:
Yes its the number of walkers each crowd has.